executor/partition_table_test.go

```diff
@@ -3089,3 +3089,15 @@ func (s *partitionTableSuite) TestIssue26251(c *C) {
 		c.Fail()
 	}
 }
+
+func (s *partitionTableSuite) TestIssue29282(c *C) {
```

Comment:
You can add a case that "for update" is explicitly used, for example `select * from t1 for update union select * from tmp1`